Lib/test/test_logging.py

```diff
@@ -4023,11 +4023,11 @@ def test_exception(self):
 
     def test_log_invalid_level_with_raise(self):
         with support.swap_attr(logging, 'raiseExceptions', True):
-            self.assertRaises(TypeError, self.logger.log, '10', 'test message')
+            self.assertRaises(TypeError, self.logger.log, 'xx', 'test message')
```

Comment:
Don't change existing tests. Just add new tests.

Reply:
'10' == 10 so I would need to remove this test. What we should be testing is if it's an unknown string, not an integer in hiding.

Reply:
No, in Python '10' != 10.

If there is a special test for '10', thus it is rejected for purpose. Until the design decision is not changed, the test shouldn't be changed.

Reply:
as far as logging levels is concerned '10' == 10. Yes I'm aware that Int != Str. It pointlessly pedantic to throw an exception for '10' when it is trivially re-cast as an Int. It's not an error worth forcing the user to go fix. "Liberal in what you accept. Strict in what you emit".

Reply:
Currently this is considered as an error. There are purposed tests for this. If you want to change this and have use cases, please open a special issue on the tracker.